changelog/5125.removal.rst

```diff
@@ -0,0 +1,2 @@
+Introduce the ``pytest.ExitCode`` Enum and make session.exitcode an instance of it.
+User defined exit codes are still valid, but consumers need to take the enum into account.
```

Comment:
"but consumers need to take the enum into account."

Not clear to me what this means, what consumers would need to do specifically?

Reply:
users of that api, like the xdist release you just made to support it

Reply:
Ahh OK, I got confused because I thought the first and second sentence were related, but the "but consumers take the enum into account" phrase was actually referring to the "Introduce the pytestExitCode" phase.

Reply:
How about:

```rst
``Session.exitcode`` values are now coded in ``pytest.ExitCode``, an ``IntEnum``. This makes the exit code available for consumer code and are more explicit other than just documentation. User defined exit codes are still valid, but should be used with caution.

The team doesn't expect this change to break test suites or plugins in general, except in esoteric/specific scenarios.

**pytest-xdist** users should upgrade to ``1.29.0`` or later, as ``pytest-xdist`` required a compatibility fix because of this change.
```

(can't use the "suggestion" feature as this spams multiple lines)